src/charm.py

```diff
@@ -528,19 +521,19 @@ def _configure(self):  # noqa: C901
             v12_migration_date=v12_migration_date,
         ).build()
 
-        restart = self._update_config(config) or restart
+        restart = self._update_config(config)
+        new_layer = self._loki_pebble_layer
```

Comment:
we don't quite know or care if it's `new`, do we?
```suggestion
        self._loki_container.add_layer(self._name, self._loki_pebble_layer, combine=True)
```